### PR TITLE
nginx: update nginxMainline and nginxModules

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update nginxMainline to 1.19.2
Update nginxModules:
 - brotli to 2020.04.23
 - develkit to v0.3.1
 - echo to v0.62
 - cache-purge to 2.5.1
 - http_proxy_connect_module to 2020.07.24
 - modsecurity-nginx to v1.0.1
 - naxsi to 2020.08.15
 - opentracing to 2020.08.15
 - pam to v1.5.2
 - push-stream to 2020.05.03
 - shibboleth to 2020.08.12
 - subsFilter to 2019.08.06
 - upstream-check to 2019.11.03
 - statsd - change mirror

Changed in nginxModules name `source` to nginxModules name.

Result:
```
./result/bin/nginx -V 2>&1 | sed -e 's|--|\n--|g'
...
--add-module=/nix/store/6pb7j6kymf3y4xs5blp3g8mwin2j22kk-dav
--add-module=/nix/store/y39g23fn8ikzcd1iy3b1bclqwjk2qmxd-moreheaders

```

cc @Mic92 @aanderse @ajs124

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ajs124** [comment]
Why rename fastcgi-cache-purge to cache-purge?
And what do we gain by introducing the name attribute?

Besides that, looks reasonable, but I can't test most of these modules.

**Comment by Izorkin** [comment]
> Why rename fastcgi-cache-purge to cache-purge?

Original name `ngx_cache_purge` - `https://github.com/nginx-modules/ngx_cache_purge#about`. This module also caches proxy, scgi, uwsgi

> And what do we gain by introducing the name attribute?

Easy to view what modules nginx is built with.

**Comment by ajs124** [comment]
> Original name `ngx_cache_purge`

Fair enough. This still breaks backwards compatibility without any warnings though. Idk how `nginxModules` interacts with something like `aliases.nix`, but this change should definitely be documented somewhere and/or throw some warning/error like "fastcgi-cache-purge was renamed to cache-purge", because configs which use this will just break otherwise.

> Easy to view what modules nginx is built with.

You mean because of the `--add-module` lines? Alright.

**Comment by Izorkin** [comment]
> > Original name `ngx_cache_purge`
> 
> Fair enough. This still breaks backwards compatibility without any warnings though. Idk how `nginxModules` interacts with something like `aliases.nix`, but this change should definitely be documented somewhere and/or throw some warning/error like "fastcgi-cache-purge was renamed to cache-purge", because configs which use this will just break otherwise.
> 
> > Easy to view what modules nginx is built with.
> 
> You mean because of the `--add-module` lines? Alright.

Added to release-notes. Variant with `aliases.nix` not working. Error: 
```
error: Alias nginxModules is still in all-packages.nix
```



> You mean because of the `--add-module` lines? Alright.

Yes.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -326,6 +326,8 @@ mapAliases ({
   neutral-style = throw "neural-style has been removed, as the upstream project has been abandoned"; # added 2020-03-28
   nfsUtils = nfs-utils;  # added 2014-12-06
   nginxUnstable = nginxMainline; # added 2018-04-25
+  nginxModules.fastcgi-cache-purge = nginxModules.cache-purge; # added 2020.08.16
+  nginxModules.ngx_aws_auth = nginxModules.aws-auth; # added 2020.08.16
```
Comment (Mic92) [inline pkgs/top-level/aliases.nix:330]: I don't think this works for nested attrsets. At least it does not work for `pythonPackages`.
You could try to add a `throw` statement within `nginxModules` so.

Comment (Izorkin) [reply pkgs/top-level/aliases.nix:330]: With `throw` not worked. Removed from `aliases`.

Comment (Mic92) [reply pkgs/top-level/aliases.nix]: Just to confirm adding a `throw` for the old attribute name in `pkgs/servers/http/nginx/modules.nix` did not work?

Comment (Izorkin) [reply pkgs/top-level/aliases.nix]: Thanks, with this variant worked:
```
error: module fastcgi-cache-purge renamed to cache-purge
```

**Comment by Izorkin** [comment]
Attribute `meta.broken = true;` does not work. How to mark non-working modules?

File: pkgs/servers/http/nginx/modules.nix [hunk 2]
```diff
@@ -18,12 +19,26 @@ let
 in
 
 {
+  fastcgi-cache-purge = throw "module fastcgi-cache-purge renamed to cache-purge";
+  ngx_aws_auth = throw "module fastcgi-cache-purge renamed to aws-auth";
```
Comment (Mic92) [inline pkgs/servers/http/nginx/modules.nix:23]: Nitpick: Just to make it a sentence:

```suggestion
  fastcgi-cache-purge = throw "fastcgi-cache-purge was renamed to cache-purge";
  ngx_aws_auth = throw "fastcgi-cache-purge was renamed to aws-auth";
```

Comment (Izorkin) [reply pkgs/servers/http/nginx/modules.nix]: Fixed.

**Comment by Mic92** [comment]
> Attribute `meta.broken = true;` does not work. How to mark non-working modules?

you can add an `assert` in the module itself however given that nginx modules are passed at build time rather than being independent derivations there is no `meta` or `meta.broken`.

**Comment by Izorkin** [comment]
I have no variant on how best to do.

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 95573` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
<br>- devpi-client
<br>- devpi-server
</details>
<details>
  <summary>4 packages built:</summary>
<br>- nginx
<br>- nginxMainline (nginxUnstable)
<br>- nginxShibboleth
<br>- tengine
</details>

**Comment by Mic92** [comment]
devpi was already broken before.

**Comment by makefu** [comment]
can 

> devpi was already broken before. cc @makefu

can confirm (related to python requests), no blocker from my side.

**Comment by ajs124** [comment]
There's still no alias/exception/anything for the module renames, so now their config will just be broken.

**Comment by Izorkin** [comment]
When using the old name, a warning will be message:
```
nix build -f ./default.nix nginxMainline
error: fastcgi-cache-purge was renamed to cache-purge
```

**Pull request merged**